### PR TITLE
Adding ClusterDuckType CRD and e2e smoke tests.

### PR DESCRIPTION
**Pull request by n3wscott**
following the documentation, results in:

```shell
(╯°□°)╯︵  ./test/kind/e2e-tests.sh
Starting End-to-End tests for Discovery API
=== RUN   TestSmoke
2020/07/23 15:25:25 new files in /var/folders/vt/3p_gcfwx1xx4_17_y1cb4jjw0000gp/T/processed_yaml595622306
I0723 15:25:25.098544   18959 manifest.go:61] Reading YAML filename/var/folders/vt/3p_gcfwx1xx4_17_y1cb4jjw0000gp/T/processed_yaml595622306
I0723 15:25:25.101475   18959 manifest.go:84] Creatingtypediscovery.knative.dev/v1alpha1, Kind=DuckTypenameminimums.smoke.knative.dev
2020/07/23 15:25:25 Created:
2020/07/23 15:25:25 {DuckType rigging-zzncomdt minimums.smoke.knative.dev  discovery.knative.dev/v1alpha1  }
    TestSmoke: smoke_test.go:34: Created a new testing rig at namespace rigging-zzncomdt.
I0723 15:25:25.144951   18959 manifest.go:123] Deleting type discovery.knative.dev/v1alpha1, Kind=DuckTypename minimums.smoke.knative.dev
--- PASS: TestSmoke (5.14s)
PASS
ok  	knative.dev/discovery/test/e2e	5.240s
```

### CODE REVIEW

**Comment by n3wscott** [comment]
/hold

I want to update the CRD

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+*/
+
+package example
+
+import (
+	"knative.dev/pkg/test/logstream"
+	"testing"
+)
+
+// TestSmoke makes sure an Addressable Service goes ready.
```
Comment (lberk) [inline test/e2e/main_test.go:26]: addressable or minimums?

File: config/300-clusterducktype.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
-  name: ducktypes.discovery.knative.dev
+  name: clusterducktypes.discovery.knative.dev
```
Comment (nachocano) [inline config/300-clusterducktype.yaml:18]: are you planning on creating a ducktype that is namespaced-scope thus you wanted a different name?
I'm wondering if it's the right thing to add info about the scope in the name?

File: test/e2e/smoke_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (nachocano) [inline test/e2e/smoke_test.go:2]: nit: 2020

File: config/300-clusterducktype.yaml (hunk 2 above)
Comment (nachocano) [reply config/300-clusterducktype.yaml:18]: I think there are some references in the README to DuckType instead of ClusterDuckType, just double check

Comment (nachocano) [reply config/300-clusterducktype.yaml:18]: if we go with ClusterDuckType, update the title of the PR

Comment (n3wscott) [reply config/300-clusterducktype.yaml:18]: I think we will also add DuckType in a followup when ClusterDuckType is working.

Comment (n3wscott) [reply config/300-clusterducktype.yaml:18]: >are you planning on creating a ducktype that is namespaced-scope thus you wanted a different name?
I'm wondering if it's the right thing to add info about the scope in the name?
yeah exactly, following the recommended pattern from kubernetes, like RoleBinding vs ClusterRoleBinding

Comment (nachocano) [reply config/300-clusterducktype.yaml:18]: ack!

**Comment by nachocano** [comment]
/approve

Let me know when you fix the conflicts and I'll lgtm

**Comment by n3wscott** [comment]
Updating with the comments, one sec.. :D thanks for the review

**Pull request merged**